src/services/RebalanceHLPv2Service.sol

```diff
@@ -0,0 +1,371 @@
+        vaultStorage.removeHLPLiquidityOnHold(_depositParam.longToken, _depositParam.longTokenAmount);
+        // Push long token to gmxV2DepositVault.
+        vaultStorage.pushToken(_depositParam.longToken, gmxV2DepositVault, _depositParam.longTokenAmount);
+      }
+      if (_depositParam.shortTokenAmount > 0) {
+        // If deploying short token, accounted as on hold and remove from HLP.
+        vaultStorage.removeHLPLiquidityOnHold(_depositParam.shortToken, _depositParam.shortTokenAmount);
+        // Push short token to gmxV2DepositVault.
+        vaultStorage.pushToken(_depositParam.shortToken, gmxV2DepositVault, _depositParam.shortTokenAmount);
+      }
+
+      // Taken WETH from caller and send to gmxV2DepositVault for execution fee
+      weth.safeTransferFrom(msg.sender, gmxV2DepositVault, _executionFee);
+      // Create a deposit order
+      _gmxOrderKey = gmxV2ExchangeRouter.createDeposit(
+        IGmxV2ExchangeRouter.CreateDepositParams({
+          receiver: address(this),
+          callbackContract: address(this),
+          uiFeeReceiver: address(0),
+          market: _depositParam.market,
+          initialLongToken: _depositParam.longToken,
+          initialShortToken: _depositParam.shortToken,
+          longTokenSwapPath: new address[](0),
+          shortTokenSwapPath: new address[](0),
+          minMarketTokens: _depositParam.minMarketTokens,
+          shouldUnwrapNativeToken: false,
+          executionFee: _executionFee,
+          callbackGasLimit: _depositParam.gasLimit
+        })
+      );
+      // Update returner
+      _gmxOrderKeys[i] = _gmxOrderKey;
+      // Keep track of deposit history
+      depositHistory[_gmxOrderKey] = _depositParam;
+
+      emit LogDepositCreated(_gmxOrderKey, _depositParam);
+    }
+  }
+
+  /// @notice Create withdraw orders on GMXv2 to rebalance HLP
+  /// @dev Caller must approve WETH to this contract
+  /// @param _withdrawParams Array of WithdrawParams
+  /// @param _executionFee Execution fee in WETH
+  function createWithdrawalOrders(
+    WithdrawalParams[] calldata _withdrawParams,
+    uint256 _executionFee
+  ) external onlyWhitelisted returns (bytes32[] memory _gmxOrderKeys) {
+    uint256 _withdrawParamsLen = _withdrawParams.length;
+    _gmxOrderKeys = new bytes32[](_withdrawParamsLen);
+
+    WithdrawalParams memory _withdrawParam;
+    bytes32 _gmxOrderKey;
+    for (uint256 i = 0; i < _withdrawParamsLen; ) {
+      _withdrawParam = _withdrawParams[i];
+
+      // Remove GM(x), accounted as on hold, and send to gmxV2WithdrawalVault.
+      vaultStorage.removeHLPLiquidityOnHold(_withdrawParam.market, _withdrawParam.amount);
+      vaultStorage.pushToken(_withdrawParam.market, gmxV2WithdrawalVault, _withdrawParam.amount);
+
+      // Taken WETH from caller and send to gmxV2WithdrawalVault for execution fee
+      weth.safeTransferFrom(msg.sender, gmxV2WithdrawalVault, _executionFee);
```

Comment:
Is it possible do one transfer before the loop? `_withdrawParamsLen * _executionFee`

Reply:
Cannot because GMXv2 will recognize the whole amount as execution fee if transfer in one go.